packages/mdc-toolbar/README.md

```diff
@@ -136,6 +290,98 @@ The provided modifiers are:
 
 | Class                                | Description                             |
 | -------------------------------------| --------------------------------------- |
-| `mdc-toolbar--fixed`                 | Make toolbar fixed to top of screen.    |
+| `mdc-toolbar--fixed`                 | Makes toolbar fixed on top and have persistent elavation |
+| `mdc-toolbar--waterfall`             | Removes fixed toolbar persistent elavation and makes it to gain elevation when a user begins to scroll down the page |
+| `mdc-toolbar--fixed-lastrow-only`    | Makes only last row of fixed toolbar anchored on top |
+| `mdc-toolbar--flexible`              | Makes toolbar first row has flexible space |
 | `mdc-toolbar__section--align-start`  | Makes section aligns to the start.      |
 | `mdc-toolbar__section--align-end`    | Makes section aligns to the end.        |
+
+
+## JS Usage
+
+### Including in code
+
+#### ES2015
+
+```javascript
+import {MDCToolbar, MDCToolbarFoundation} from 'mdc-toolbar';
+```
+
+#### CommonJS
+
+```javascript
+const mdcToolbar = require('mdc-toolbar');
+const MDCToolbar = mdcToolbar.MDCToolbar;
+const MDCToolbarFoundation = mdcToolbar.MDCToolbarFoundation;
+```
+
+#### AMD
+
+```javascript
+require(['path/to/mdc-toolbar'], mdcToolbar => {
+  const MDCToolbar = mdcToolbar.MDCToolbar;
+  const MDCToolbarFoundation = mdcToolbar.MDCToolbarFoundation;
+});
+```
+
+#### Global
+
+```javascript
+const MDCToolbar = mdc.toolbar.MDCToolbar;
+const MDCToolbarFoundation = mdc.toolbar.MDCToolbarFoundation;
+```
+
+### Automatic Instantiation
+
+If you do not care about retaining the component instance for the toolbar, simply call `attachTo()`
+and pass it a DOM element.
+
+```javascript
+mdc.toolbar.MDCToolbar.attachTo(document.querySelector('.mdc-toolbar'));
+```
+
+### Manual Instantiation
+
+```javascript
+import {MDCToolbar} from 'mdc-toolbar';
+
+const toolbar = new MDCToolbar(document.querySelector('.mdc-toolbar'));
+```
+
+### Using the MDCToolbar Foundation Class
+
+#### API
+
+| Method Signature | Description |
+| --- | --- |
+| `updateAdjustElementStyles() => void` | sets `AdjustElement` proper `margin-top`. |
+
+
+#### Event
+
+| Event Name | Event Data Structure | Description |
+| --- | --- | --- |
+| `change` | {flexibleExpansionRatio: number} | Emits the ratio of current flexible space to total flexible space height. So when it is minimized, ratio equals to 0 and when it is maximized, ratio equals to 1. |
```

Comment:
supernit: should we wrap "{flexibleExpansionRatio: number}" in backticks so it gets formatted as code?